### PR TITLE
applications: nrf_desktop: BLE latency handling

### PR DESCRIPTION
**Pull request by MarekPieta**
Change moves BLE latency handling to separate module.
BLE latency is set on `config_event` or `config_fetch_request_event`.

### CODE REVIEW

**Comment by MarekPieta** [comment]
@ppryga, could you review ?

File: applications/nrf_desktop/src/modules/ble_latency.c [hunk 1]
```diff
@@ -42,6 +53,88 @@ static void security_timeout_fn(struct k_work *w)
...
+
+	if (!err || (err == -EALREADY)) {
+		if (low_latency) {
+			latency_state |= LOW_LATENCY_ENABLED;
+		} else {
+			latency_state &= ~LOW_LATENCY_ENABLED;
+		}
+
+		if (err == -EALREADY) {
+			LOG_INF("Conn parameters are already updated");
```
Comment (pdunaj) [inline applications/nrf_desktop/src/modules/ble_latency.c]: s/are/were

File: applications/nrf_desktop/src/modules/ble_latency.c [hunk 2]
```diff
@@ -87,6 +182,21 @@ static bool event_handler(const struct event_header *eh)
 		return false;
 	}
 
+	if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE) &&
+	    (is_config_event(eh) || is_config_fetch_request_event(eh))) {
+		if (!active_conn) {
+			return false;
+		}
+
+		if (likely(latency_state & LOW_LATENCY_ENABLED)) {
```
Comment (pdunaj) [inline applications/nrf_desktop/src/modules/ble_latency.c]: `likely` is not needed

File: applications/nrf_desktop/src/modules/ble_latency.c (hunk 1 above)
Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/ble_latency.c]: done

File: applications/nrf_desktop/src/modules/ble_latency.c [hunk 3]
```diff
@@ -42,6 +53,88 @@ static void security_timeout_fn(struct k_work *w)
...
+			latency_state |= LOW_LATENCY_ENABLED;
+		} else {
+			latency_state &= ~LOW_LATENCY_ENABLED;
+		}
+
+		if (err == -EALREADY) {
+			LOG_INF("Conn parameters are already updated");
+		}
+
+	} else if (err) {
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/ble_latency.c]: removed

File: applications/nrf_desktop/src/modules/ble_latency.c [hunk 4]
```diff
@@ -42,6 +53,88 @@ static void security_timeout_fn(struct k_work *w)
...
+		LOG_WRN("Failed to update conn parameters (err %d)", err);
+		return;
+	}
+
+	LOG_INF("BLE latency %screased", low_latency ? "de" : "in");
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/ble_latency.c]: done

File: applications/nrf_desktop/src/modules/ble_latency.c (hunk 2 above)
Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/ble_latency.c]: Ok, removed

File: applications/nrf_desktop/src/modules/ble_latency.c [hunk 5]
```diff
@@ -42,6 +53,88 @@ static void security_timeout_fn(struct k_work *w)
...
+				      LOW_LATENCY_CHECK_PERIOD_MS);
+	} else {
+		latency_state &= ~LOW_LATENCY_ENABLED;
+		k_delayed_work_cancel(&low_latency_check);
+	}
+}
+
+static void init(void)
+{
+	static struct bt_conn_cb conn_callbacks = {
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/ble_latency.c]: `bt_conn_cb_register` function takes `struct bt_conn_cb` as argument (without `const`)

**Pull request merged**